tests/test_task.py

```diff
@@ -276,3 +278,61 @@ async def test_tool_failure(self, base_query_request: QueryRequest) -> None:
         mock_SearchIndex.assert_called(), "Expected failures to come from unit test"
         assert metrics_callback.eval_means["correct"] == 0.0
         assert metrics_callback.eval_means["correct_unsure"] == 0.0
+
+
+@pytest.mark.asyncio
+async def test_docdetails_merge_with_non_list_fields():
```

Comment:
Let's put this into `test_paperqa` next to `test_dois_resolve_to_correct_journals`

This is unrelated to the `TaskDataset` stuff